promise.go

```diff
@@ -200,6 +200,51 @@ func AwaitAll(promises ...*Promise) {
 	}
 }
 
+func All(promises []*Promise) *Promise {
```

Comment:
Good overall!
The only concern I have is that you are creating additional goroutines for every promise. I believe this is unnecessary and adds overhead.
Please have a look at  [dev branch implementation](https://github.com/chebyrash/promise/blob/dev/promise.go#L201) where I use native Then and Catch methods

Reply:
Aha, you're correct especially since each promise already launches it's executor in a subroutine, mhm gotcha, will update code.

Reply:
Yeah :)

Reply:
Yeah saw that :), look at my latest update though